### PR TITLE
Release 2.17.0

### PR DESCRIPTION
**Pull request by pallavisontakke**
This release adds support for PostgreSQL 17, significantly improves the performance of continuous aggregate refreshes,
and contains performance improvements for analytical queries and delete operations over compressed hypertables.
We recommend that you upgrade at the next available opportunity.

**Highlighted features in TimescaleDB v2.17.0**

* Full PostgreSQL 17 support for all existing features. TimescaleDB v2.17 is available for PostgreSQL 14, 15, 16, and 17.

* Significant performance improvements for continuous aggregate policies: continuous aggregate refresh is now using
`merge` instead of deleting old materialized data and re-inserting.

  This update can decrease dramatically the amount of data that must be written on the continuous aggregate in the
  presence of a small number of changes, reduce the `i/o` cost of refreshing a continuous aggregate, and generate fewer
  Write-Ahead Logs (`WAL`).
  Overall, continuous aggregate policies will be more lightweight, use less system resources, and complete faster.

* Increased performance for real-time analytical queries over compressed hypertables:
  we are excited to introduce additional Single Instruction, Multiple Data (`SIMD`) vectorization optimization to our
  engine by supporting vectorized execution for queries that group by using the `segment_by` column(s) and
  aggregate using the basic aggregate functions (`sum`, `count`, `avg`, `min`, `max`).

  Stay tuned for more to come in follow-up releases! Support for grouping on additional columns, filtered aggregation,
  vectorized expressions, and `time_bucket` is coming soon.

* Improved performance of deletes on compressed hypertables when a large amount of data is affected.

  This improvement speeds up operations that delete whole segments by skipping the decompression step.
  It is enabled for all deletes that filter by the `segment_by` column(s).

**PostgreSQL 14 deprecation announcement**

  We will continue supporting PostgreSQL 14 until April 2025. Closer to that time, we will announce the specific
  version of TimescaleDB in which PostgreSQL 14 support will not be included going forward.

**Features**
* #6882: Allow delete of full segments on compressed chunks without decompression.
* #7033: Use `merge` statement on continuous aggregates refresh.
* #7126: Add functions to show the compression information.
* #7147: Vectorize partial aggregation for `sum(int4)` with grouping on `segment by` columns.
* #7204: Track additional extensions in telemetry.
* #7207: Refactor the `decompress_batches_scan` functions for easier maintenance.
* #7209: Add a function to drop the `osm` chunk.
* #7275: Add support for the `returning` clause for `merge`.
* #7200: Vectorize common aggregate functions like `min`, `max`, `sum`, `avg`, `stddev`, `variance` for compressed columns
  of arithmetic types, when there is grouping on `segment by` columns or no grouping.

**Bug fixes**
* #7187: Fix the string literal length for the `compressed_data_info` function.
* #7191: Fix creating default indexes on chunks when migrating the data.
* #7195: Fix the `segment by` and `order by` checks when dropping a column from a compressed hypertable.
* #7201: Use the generic extension description when building `apt` and `rpm` loader packages.
* #7227: Add an index to the `compression_chunk_size` catalog table.
* #7229: Fix the foreign key constraints where the index and the constraint column order are different.
* #7230: Do not propagate the foreign key constraints to the `osm` chunk.
* #7234: Release the cache after accessing the cache entry.
* #7258: Force English in the `pg_config` command executed by `cmake` to avoid the unexpected building errors.
* #7270: Fix the memory leak in compressed DML batch filtering.
* #7286: Fix the index column check while searching for the index.
* #7290: Add check for null offset for continuous aggregates built on top of continuous aggregates.
* #7301: Make foreign key behavior for hypertables consistent.
* #7318: Fix chunk skipping range filtering.
* #7320: Set the license specific extension comment in the install script.

**Thanks**
* @MiguelTubio for reporting and fixing the Windows build error.
* @posuch for reporting the misleading extension description in the generic loader packages.
* @snyrkill for discovering and reporting the issue with continuous aggregates built on top of continuous aggregates.

Disable-check: force-changelog-file
Disable-check: commit-count

### CODE REVIEW

File: CHANGELOG.md
Comment (atovpeko) [inline CHANGELOG.md:17]: could you please use lower case here, we're trying to use it consistently for feature names

File: CHANGELOG.md
Comment (atovpeko) [inline CHANGELOG.md:24]:
```suggestion
**Bug fixes**
```

**Review by atovpeko — CHANGES_REQUESTED** [review]
Just very minor things. I see that both bugfixes and bug fixes are used, so I suggest going consistently with bug fixes.

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: edited so.

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: edited so

File: CHANGELOG.md
Comment (atovpeko) [inline CHANGELOG.md:20]:
```suggestion
We will continue supporting PostgreSQL 14 until April 2025. Closer to that time, we will announce the specific version of TimescaleDB in which PostgreSQL 14 support will not be included going forward.
```

**Review by atovpeko — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:31]: @pallavisontakke @svenklemm @mkindahl Why is this included? Should we be careful about releasing anything related to the TAM early?

For sure it should not be part of the list in the release notes even if we want to keep it around

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:23]: This sounds really deceptive - let's rephrase a little bit

```suggestion
* #6882: Allow delete of fully covered compressed batches on compressed chunks without decompression.
```

@svenklemm agree?

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:21]:
```suggestion
This release adds support for PostgreSQL 17, significantly improves the performance of continuous aggregate refreshes, 
and contains performance improvements for analytical queries and delete operations over compressed hypertables. 
We recommend that you upgrade at the next available opportunity.

**Highlighted features in TimescaleDB v2.17.0**
* Full PostgreSQL 17 support for all existing features. TimescaleDB v2.17 is available for PostgreSQL 14, 15, 16 and 17.

  PostgreSQL 17 support for the Toolkit Extension will follow soon and will be announced on [Toolkit's release notes](https://github.com/timescale/timescaledb-toolkit/releases).  

* Significant performance improvements for continuous aggregate policies: Continuous aggregate refresh is now using MERGE instead of deleting old materialized data and re-inserting.

  This update can decrease dramatically the amount of data that must be written on the continuous aggregate in the
  presence of a small number of changes, reduce the I/O cost of refreshing a continuous aggregate and generate less WAL.
  Overall, continuous aggregate policies will be more lightweight, use less system resources and complete faster.

* Increased performance for real-time analytical queries over compressed hypertables:
  We are excited to introduce vectorized query execution (SIMD optimizations) to our engine by supporting vectorized
  execution for queries that group by using the `segment_by` column(s) and aggregate using the basic aggregate
  functions (sum, count, avg, min/max).  

  Stay tuned for more to come in follow-up releases! Support for grouping on additional columns, filtered aggregation,
  vectorized expressions and `time_bucket` is coming soon. 

* Improved performance of deletes on compressed hypertables when a large amount of data is affected.
This improvement speeds up operations that delete everything for a specific segment or when whole compressed 
batches can be deleted (of up to 1000 continuous records when ordered by time or the order_by). 

**PostgreSQL 14 deprecation announcement**
We will continue supporting PostgreSQL 14 until April 2025. Sooner to that time, we will announce the specific
version of TimescaleDB in which PostgreSQL 14 support will not be included going forward.

```

Comment (iroussos) [reply CHANGELOG.md:21]: @fabriziomello @akuzm @svenklemm for your review in case anything needs fixing or I forgot about something important.

**Review by iroussos — CHANGES_REQUESTED** [review]
Thank you @pallavisontakke, I have added my first round of comments and suggestions

File: CHANGELOG.md
Comment (mkindahl) [reply CHANGELOG.md:31]: Agree, we can exclude this commit from the release. It is needed for the TAM eventually, but the code does not do much currently (except allowing multiple commands to `ALTER TABLE`).

**Comment by iroussos** [comment]
@ramonguiu @josesahad for your review of the following deprecation, This follows our cadence of deprecating one version each time we add support for another. Otherwise we'll be stuck with supporting 4 major PG versions at once for the rest of the year. cc @philkra 

```
**PostgreSQL 14 deprecation announcement**
We will continue supporting PostgreSQL 14 until April 2025. Closer to that time, we will announce the specific
version of TimescaleDB in which PostgreSQL 14 support will not be included going forward.
```

More info and additional stats on the distribution of the releases in Timescale Cloud on the following [internal thread](https://iobeam.slack.com/archives/C07LE35HUQM/p1727177665147409).

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md:31]: Does that mean we want to revert just this release-note while keeping the src-code commit in `main` ? As this description in-detail would get included during `2.18.0` hypercore  release.

Comment (mkindahl) [reply CHANGELOG.md:31]: The commit has to be in main for a later release, but when you create the 2.17.x branch, you can remove this commit from the history. It contains a name string which might not be the final name of the table access method, so it makes sense to remove it.

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: edited so.

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: Edited, please check.

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: Edited, please check.

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:25]: Your call, but both I/O and [WAL](https://www.postgresql.org/docs/current/wal-intro.html) are canonically referenced in capital letters

```suggestion
  presence of a small number of changes, reduce the `I/O` cost of refreshing a continuous aggregate and generate less
  amount of Write-Ahead Logs (`WAL`).
```

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:29]: Also [Single instruction, multiple data (SIMD)](https://en.wikipedia.org/wiki/Single_instruction,_multiple_data) is usually referenced using capital letters. Yours once more, but it looks weird to me:

```suggestion
  We are excited to introduce vectorized query execution (`SIMD` optimizations) to our engine by supporting vectorized
```

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:31]: If we are going to mark those, why not mark everything? 

```suggestion
  functions (`sum`, `count`, `avg`, `min`, `max`).
```

**Review by iroussos — APPROVED** [review]
Thank you, looks good to me - I just added a few final small suggestions

**Review by atovpeko — COMMENTED** [review]
Sorry, a bit more nitpicking.

File: CHANGELOG.md
Comment (svenklemm) [reply CHANGELOG.md]: Hmm not sure your wording makes it more clear what this is about. This is about deleting full segments without decompression.

File: CHANGELOG.md
Comment (svenklemm) [reply CHANGELOG.md:18]: Do we need status update on toolkit on the timescaledb release notes?

Comment (pallavisontakke) [reply CHANGELOG.md:18]: Question for @iroussos

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: Added Fabrizio & Alex as reviewers to this PR. Please check.

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: Edited, please check

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: Edited so.

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: Edited so.

File: CHANGELOG.md
Comment (atovpeko) [inline CHANGELOG.md:47]:
```suggestion
* #6882: Allow delete of fully covered compressed batches on compressed chunks,  that is, delete full segments without decompression.
```

**Review by atovpeko — APPROVED** [review]
Approving with just 1 minor fix. Thanks!

File: CHANGELOG.md
Comment (iroussos) [reply CHANGELOG.md:18]: Yeah we don't usually include that info, I am going to remove it

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:18]:
```suggestion
```

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:47]:
```suggestion
* #6882: Allow delete of full segments on compressed chunks without decompression.
```

File: CHANGELOG.md
Comment (iroussos) [inline CHANGELOG.md:39]:
```suggestion
  This improvement speeds up operations that delete whole segments by skipping the decompression step.
  It is enabled for all deletes that filter by the `segment_by` column(s).
```

**Review by iroussos — APPROVED** [review]
Thank you @pallavisontakke, added my final fixes - I will directly commit them and we should be good on my end

File: CHANGELOG.md
Comment (iroussos) [reply CHANGELOG.md]: Resolving this conversation - please check directly on the final version of the changelog

**Review by philkra — APPROVED** [review]
LGTM

**Review by svenklemm — APPROVED** [review]

**Review by mkindahl — APPROVED** [review]

File: CHANGELOG.md
Comment (pallavisontakke) [reply CHANGELOG.md]: Removed the release-note for PR #7295 , as it's more a part of 2.18 story.

File: sql/updates/2.17.0--2.16.1.sql
Comment (fabriziomello) [inline sql/updates/2.17.0--2.16.1.sql:3]: The downgrade script should include all from `sql/updates/reverse-dev.sql`.

Comment (svenklemm) [reply sql/updates/2.17.0--2.16.1.sql:3]: this is the pr into main which will not have the downgrade changes

File: version.config
Comment (svenklemm) [inline version.config:3]:
```suggestion
version = 2.18.0-dev
update_from_version = 2.17.0
downgrade_to_version = 2.16.1
```

Comment (svenklemm) [reply version.config:3]: This goes into main so the new version for main will be 2.18.0-dev

**Review by svenklemm — CHANGES_REQUESTED** [review]
Since this is the main PR the reverse-dev changes should not be part of this PR

File: sql/updates/2.17.0--2.16.1.sql
Comment (pallavisontakke) [reply sql/updates/2.17.0--2.16.1.sql:119]: Edited so.

File: version.config
Comment (pallavisontakke) [reply version.config]: Edited so.

File: sql/CMakeLists.txt
Comment (svenklemm) [inline sql/CMakeLists.txt:49]: this needs to be removed from pr against main

File: sql/CMakeLists.txt
Comment (svenklemm) [inline sql/CMakeLists.txt:50]:
```suggestion
set(CURRENT_REV_FILE reverse-dev.sql)
```

**Review by svenklemm — APPROVED** [review]

**Pull request merged**
